mgmt/rest/v2/plugin.go

```diff
@@ -244,13 +280,40 @@ func (s *apiV2) loadPlugin(w http.ResponseWriter, r *http.Request, _ httprouter.
 			default:
 				ec = 500
 			}
+			cleanUpTempFiles(rp)
 			Write(ec, rb, w)
 			return
 		}
+		defer cleanUpTempFiles(rp)
 		Write(201, catalogedPluginBody(r.Host, pl), w)
 	}
 }
 
+func isTLSEnabled(cert, key string) bool {
+	if cert != "" && key != "" {
+		return true
+	}
+	return false
+}
+
+func createTempFile(b []byte, fn string) (string, error) {
+	file, err := ioutil.TempFile(os.TempDir(), fn)
+	if err != nil {
+		return "", err
+	}
+	_, err = file.Write(b)
+	if err != nil {
+		return "", err
+	}
+	return file.Name(), nil
+}
+
+func cleanUpTempFiles(rp *core.RequestedPlugin) {
```

Comment:
@marcintao, are you suggesting that we only provide the paths. If they installed snap-cli on a different machine/remote, it's their issue to resolve it? Please confirm.